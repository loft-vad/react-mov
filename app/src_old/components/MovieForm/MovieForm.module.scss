.wrapper {
  display: grid;
  grid-template-columns: 2fr 1fr;
  gap: 30px;

  input,
  select,
  textarea {
    width: 100%;
    border: none;
    background: rgba(50, 50, 50, 0.948044);
    mix-blend-mode: normal;
    opacity: 0.8;
    border-radius: 4px;
    font-weight: 400;
    font-size: 20px;
    line-height: 24px;

    color: var(--white);
    padding: 17px 18px 16px;

    &::placeholder {
      font-weight: 400;
      font-size: 20px;
      line-height: 24px;

      color: var(--white);

      mix-blend-mode: normal;
      opacity: 0.3;
    }
  }

  .inputWrapper {
    label {
      padding-bottom: 13px;
      font-weight: 600;
      font-size: 16px;
      line-height: 20px;
      letter-spacing: 0.888889px;
      text-transform: uppercase;
      color: var(--main);
      mix-blend-mode: normal;
      opacity: 0.8;
      display: block;
    }

    &.fullWidth {
      grid-column: 1 / 3;
    }
  }

}

.formActions {
  width: 100%;
  display: flex;
  gap: 13px;
  justify-content: flex-end;
  padding-top: 60px;
}