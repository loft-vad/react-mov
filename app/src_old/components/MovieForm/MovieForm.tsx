import styles from "./MovieForm.module.css";
import Button from "../common/Button/Button";
import { MovieFull } from "../MovieDetails/MovieDetails";
import { useState } from "react";

interface MovieFormProps {
  movieInfo?: MovieFull;
}

const MovieForm: React.FC<MovieFormProps> = ({
  movieInfo = {
    title: "",
    rating: 0,
    duration: 0,
    overview: "",
    id: 0,
    poster_path: "",
    releaseYear: 0,
    genres: ["All"],
  },
}): React.ReactElement => {
  const [formState, setFormState] = useState<MovieFull>(movieInfo);

  const formSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
    event.preventDefault();
    var data = new FormData(event.target);
    let formObject = Object.fromEntries(data.entries());
    console.log(formObject);
    console.log(formState);
  };

  const handleChange = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>,
  ) => {
    const { name, value } = event.target;

    setFormState((prevValues) => ({
      ...prevValues,
      [name]: value,
    }));
  };

  return (
    <form onSubmit={formSubmit} role="form">
      <div className={styles.wrapper}>
        <div className={styles.inputWrapper}>
          <label htmlFor="title">Title</label>
          <input
            id="title"
            name="title"
            type="text"
            placeholder=""
            value={formState.title}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="releaseYear">Release Date</label>
          <input
            id="releaseYear"
            name="releaseYear"
            type="date"
            placeholder="Select Date"
            min="1997"
            max="2222"
            step="1"
            value={formState.releaseYear}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="url">Movie Url</label>
          <input
            id="url"
            name="url"
            type="text"
            placeholder=""
            value={formState.poster_path}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="rating">Rating</label>
          <input
            id="rating"
            name="rating"
            type="number"
            placeholder=""
            value={formState.rating}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="genre">Genre</label>
          <select id="genre" name="genre" value={formState.genres[0]} onChange={handleChange}>
            {formState.genres.map((genre) => (
              <option key={genre} value={genre}>
                {genre}
              </option>
            ))}
          </select>
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="duration">Runtime</label>
          <input
            id="duration"
            name="duration"
            type="number"
            placeholder=""
            value={formState.duration}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper + " " + styles.fullWidth}>
          <label htmlFor="overview">Overview</label>
          <textarea
            id="overview"
            name="overview"
            rows={5}
            placeholder="Movie description"
            value={formState.overview}
            onChange={handleChange}
          />
        </div>
      </div>
      <div className={styles.formActions}>
        <Button text="Reset" type="reset" />
        <Button text="Submit" type="submit" />
      </div>
    </form>
  );
};

export default MovieForm;
