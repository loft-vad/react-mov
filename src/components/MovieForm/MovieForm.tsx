import styles from "./MovieForm.module.scss";
import Button from "../common/Button/Button";
import { MovieFull } from "../MovieDetails/MovieDetails";
import { useState } from "react";

interface MovieFormProps {
  movieInfo?: MovieFull;
}

const MovieForm: React.FC<MovieFormProps> = ({
  movieInfo = {
    movieName: "",
    rating: 0,
    duration: 0,
    description: "",
    id: 0,
    imageUrl: "",
    releaseYear: 0,
    genres: ["All"],
  },
}): React.ReactElement => {
  const [formState, setFormState] = useState<MovieFull>(movieInfo);

  const formSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
    event.preventDefault();
    var data = new FormData(event.target);
    let formObject = Object.fromEntries(data.entries());
    console.log(formObject);
    console.log(formState);
  };

  const handleChange = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>,
  ) => {
    const { name, value } = event.target;

    setFormState((prevValues) => ({
      ...prevValues,
      [name]: value,
    }));
  };

  return (
    <form onSubmit={formSubmit} role="form">
      <div className={styles.wrapper}>
        <div className={styles.inputWrapper}>
          <label htmlFor="title">Title</label>
          <input
            id="movieName"
            name="movieName"
            type="text"
            placeholder=""
            value={formState.movieName}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="releaseYear">Release Date</label>
          <input
            id="releaseYear"
            name="releaseYear"
            type="date"
            placeholder="Select Date"
            min="1997"
            max="2222"
            step="1"
            value={formState.releaseYear}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="url">Movie Url</label>
          <input
            id="url"
            name="url"
            type="text"
            placeholder=""
            value={formState.imageUrl}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="rating">Rating</label>
          <input
            id="rating"
            name="rating"
            type="number"
            placeholder=""
            value={formState.rating}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="genre">Genre</label>
          <select id="genre" name="genre" value={formState.genres[0]} onChange={handleChange}>
            {formState.genres.map((genre) => (
              <option key={genre} value={genre}>
                {genre}
              </option>
            ))}
          </select>
        </div>
        <div className={styles.inputWrapper}>
          <label htmlFor="duration">Runtime</label>
          <input
            id="duration"
            name="duration"
            type="number"
            placeholder=""
            value={formState.duration}
            onChange={handleChange}
          />
        </div>
        <div className={styles.inputWrapper + " " + styles.fullWidth}>
          <label htmlFor="description">Overview</label>
          <textarea
            id="description"
            name="description"
            rows={5}
            placeholder="Movie description"
            value={formState.description}
            onChange={handleChange}
          />
        </div>
      </div>
      <div className={styles.formActions}>
        <Button text="Reset" type="reset" />
        <Button text="Submit" type="submit" />
      </div>
    </form>
  );
};

export default MovieForm;
