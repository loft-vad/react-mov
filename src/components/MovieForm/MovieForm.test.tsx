import React from "react";

import { render } from "@testing-library/react";
import MovieForm from "./MovieForm";

describe("MovieForm component", () => {
  it("should match snapshot", () => {
    const wrapper = render(<MovieForm />);

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });
});
