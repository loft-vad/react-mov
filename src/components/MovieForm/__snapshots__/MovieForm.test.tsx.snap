// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieForm component should match snapshot 1`] = `
<form>
  <div
    class="wrapper"
  >
    <div
      class="inputWrapper"
    >
      <label
        for="title"
      >
        Title
      </label>
      <input
        id="movieName"
        name="movieName"
        placeholder=""
        type="text"
        value=""
      />
    </div>
    <div
      class="inputWrapper"
    >
      <label
        for="releaseYear"
      >
        Release Date
      </label>
      <input
        id="releaseYear"
        max="2222"
        min="1997"
        name="releaseYear"
        placeholder="Select Date"
        step="1"
        type="date"
        value="0"
      />
    </div>
    <div
      class="inputWrapper"
    >
      <label
        for="url"
      >
        Movie Url
      </label>
      <input
        id="url"
        name="url"
        placeholder=""
        type="text"
        value=""
      />
    </div>
    <div
      class="inputWrapper"
    >
      <label
        for="rating"
      >
        Rating
      </label>
      <input
        id="rating"
        name="rating"
        placeholder=""
        type="number"
        value="0"
      />
    </div>
    <div
      class="inputWrapper"
    >
      <label
        for="genre"
      >
        Genre
      </label>
      <select
        id="genre"
        name="genre"
      >
        <option
          value="All"
        >
          All
        </option>
      </select>
    </div>
    <div
      class="inputWrapper"
    >
      <label
        for="duration"
      >
        Runtime
      </label>
      <input
        id="duration"
        name="duration"
        placeholder=""
        type="number"
        value="0"
      />
    </div>
    <div
      class="inputWrapper fullWidth"
    >
      <label
        for="description"
      >
        Overview
      </label>
      <textarea
        id="description"
        name="description"
        placeholder="Movie description"
        rows="5"
      />
    </div>
  </div>
  <div
    class="formActions"
  >
    <button
      class="button"
      type="reset"
    >
      Reset
    </button>
    <button
      class="button"
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
